### DISCUSSION TITLE
Should we release 0.66.2?

### DISCUSSION THREAD

**Discussion by lunaleaps** [reactwg/react-native-releases#3] [Patches]
~~Current Release: [0.66.1](https://github.com/facebook/react-native/releases/tag/v0.66.1), [related issue](https://github.com/react-native-community/releases/issues/254)~~
## [0.66.2 has been released](https://github.com/facebook/react-native/releases/tag/v0.66.2) -- locking this discussion and moving [discussion to 0.66.3](https://github.com/reactwg/react-native-releases/discussions/5#discussioncomment-1594605)

Conversations on this thread are limited to 0.66 releases major issues and backport (cherry-pick) requests from commits that are already on master.

An example of a good such request is a bug fix for a serious issue that has been merged into main but did not make the 0.66.1 cut, with a link to the specific commit hash on main with the commit to cherry-pick, like this example link: facebook/react-native@bd2b7d6

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.

### List of cherry picks
-  https://github.com/facebook/react-native/commit/5f7deb5f1d17050ce399dd43a86495d5c3c8ba83
-  Hide the logbox window explicitly. https://github.com/facebook/react-native/commit/72ea0e111fccd99456abf3f974439432145585e3
- fix: compare the LogBoxData ignorePatterns with the right code https://github.com/facebook/react-native/commit/a950634424cddf31c0adb6c9799adf1cc5f83bf0
### Local commits to backport to main
-

**Comment by sandy-garrido** [comment]
[facebook/react-native@c4cd447](https://github.com/facebook/react-native/commit/249b8d21c4b9734bc607ed1627055f9d4c4cd447) - [Issue  #32372](https://github.com/facebook/react-native/issues/32372) There seems to be a problem running jest, the preprocessor.js from 0.65.1 works okay but since 0.66 it has been a problem and could halt our iOS 15 support

**Reply by mark-quinn** [reply]
Having the exact same issues after upgrading from 0.63.4 to 0.66.1. Entire test suite fails.

**Reply by lunaleaps** [reply]
Can someone submit a PR with this change so we can review?

**Reply by LukePurtell** [reply]
We are also having this issue upgrading from 0.65.1 to 0.66.1. Our entire test suite fails with a similar issue that seems to stem from the jest preprocessor.

**Reply by lunaleaps** [reply]
Great! It sounds like there's a solution, we can pick it in if someone submits a PR and links it here 😄

**Reply by sandy-garrido** [reply]
> Great! It sounds like there's a solution, we can pick it in if someone submits a PR and links it here 😄

A solution hasn't been merged into main, it would be to revert the changes made to preprocessor.js on this commit [facebook/react-native@c4cd447](https://github.com/facebook/react-native/commit/249b8d21c4b9734bc607ed1627055f9d4c4cd447). I think I misunderstood how this works. I guess for this to be cherry picked, the revert would need to be merged into main already, which it's not so this fix is out of the question.

**Reply by lunaleaps** [reply]
This seems like a pretty critical issue. I think if we can get a PR with a fix, we can merge to main and pick into here. I think submitting a PR with what you needed to do to fix (maybe you don't need a full revert, etc) will help start a discussion with @rubennorte and team about what the underlying issue is with https://github.com/facebook/react-native/commit/249b8d21c4b9734bc607ed1627055f9d4c4cd447

**Reply by diegotsi** [reply]
Any news ?

**Comment by joe-sam** [comment]
[Hide the logbox window explicitly.](https://github.com/facebook/react-native/pull/32435)  #32435.
Ok, this is not as yet merged or tested but can see immediately that it a good hack that would probably be really, really super useful !!! 

[ fix: compare the LogBoxData ignorePatterns with the right code](https://github.com/facebook/react-native/pull/31977) #31977 
This has probably baked in long enough to be stable.

**Reply by lunaleaps** [reply]
We have a policy of picking in changes that are critical to developer workflow. Can you expand on why this is critical to have in for 0.66.2?

**Reply by joe-sam** [reply]
I understand that for almost all basic build processes and minimal testing apps the LogBox  may not be a necessity.

However in my view, for the vast majority developers to have confidence in an Release Candidate or Release, the runtime debugging user experience on the front end  user interface should be improved to a pleasant degree and be accurate and reliable and factor out all the redundancy, and should also  be easily configurable as well.

While not absolutely critical, these fixes do appear to align to this viewpoint.

To that end I would even go so far to suggest that LogBox  library should probably be as independent from the React-Native in the same way that all the other superfluous components are being removed from the core, but that would be the topic for a different forum.

**Reply by lunaleaps** [reply]
Picked both in

**Comment by divinghero** [comment]
from 0.63.4 ro 0.66.1，encounter this：
ios/Pods/Headers/Public/RCT-Folly/folly/math.h:26:10: 'cmath' file not found
ios/Pods/Headers/Public/React-Core/React/RCTConvert.h:17:9: Could not build module 'yoga'
<img width="980" alt="image" src="https://user-images.githubusercontent.com/5936206/138548132-02684b23-cda5-4102-9380-506f1e599eb4.png">

**Comment by togenius1** [comment]
RCT-Folly issues: React project build failed on mac M1. I have stuck for 2 weeks.

**Reply by lunaleaps** [reply]
This discussion is for cherrypicks for releasing potential patches. For your issue, can you create an [upgrade issue ](https://github.com/facebook/react-native/issues/new?assignees=&labels=Needs%3A+Triage+%3Amag%3A%2CType%3A+Upgrade+Issue&template=upgrade-regression-form.yml) or see if there are similar issues? If you have a fix, we can pick it into a new patch, thanks! Going to mark this as off-topic for now.

**Comment by kelset** [comment]
looks like we might have forgot to do the same local change we did for 0.65: https://github.com/facebook/react-native/commit/5f7deb5f1d17050ce399dd43a86495d5c3c8ba83

Unless it was intentional, this is a pretty important thing to get in - so a 0.66.2 might be needed
